include/nbl/ext/FullScreenTriangle/FullScreenTriangle.h

```diff
@@ -9,74 +9,106 @@
+					for (size_t i=0ull; i<asset::SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; i++)
+						inputParams.bindings[i] = {0u,asset::EVIR_PER_VERTEX};
+				}
+
+				auto& assemblyParams = std::get<asset::SPrimitiveAssemblyParams>(protoPipeline);
+				assemblyParams.primitiveRestartEnable = false;
+				assemblyParams.primitiveType = asset::EPT_TRIANGLE_LIST;
+				assemblyParams.tessPatchVertCount = 3u;
+
+				auto& blendParams = std::get<asset::SBlendParams>(protoPipeline);
+				blendParams.logicOpEnable = false;
+				blendParams.logicOp = nbl::asset::ELO_NO_OP;
+				for (size_t i = 0ull; i < nbl::asset::SBlendParams::MAX_COLOR_ATTACHMENT_COUNT; i++)
+					blendParams.blendParams[i].attachmentEnabled = (i == 0ull);
+
+				auto& rasterParams = std::get<asset::SRasterizationParams>(protoPipeline);
+				rasterParams.faceCullingMode = nbl::asset::EFCM_NONE;
+				rasterParams.depthCompareOp = nbl::asset::ECO_ALWAYS;
+				rasterParams.minSampleShading = 1.f;
+				rasterParams.depthWriteEnable = false;
+				rasterParams.depthTestEnable = false;
+
+				return protoPipeline;
+			}
+
+			inline core::smart_refctd_ptr<video::IGPURenderpassIndependentPipeline> createRenderpassIndependentPipeline(video::ILogicalDevice* logicalDevice, NBL_PROTO_PIPELINE& protoPipeline, core::smart_refctd_ptr<video::IGPUSpecializedShader>&& gpuFragmentShader, core::smart_refctd_ptr<video::IGPUPipelineLayout>&& pipelineLayout)
+			{
+				if (!logicalDevice)
+					assert(false);
+
+				video::IGPUSpecializedShader* gpuShaders[] = { std::get<core::smart_refctd_ptr<video::IGPUSpecializedShader>>(protoPipeline).get(), gpuFragmentShader.get() };
+				auto gpuRenderpassIndependentPipeline = logicalDevice->createGPURenderpassIndependentPipeline
+				(	
+					nullptr,
+					std::move(pipelineLayout),
+					gpuShaders,
+					gpuShaders + 2,
+					std::get<asset::SVertexInputParams>(protoPipeline),
+					std::get<asset::SBlendParams>(protoPipeline),
+					std::get<asset::SPrimitiveAssemblyParams>(protoPipeline),
+					std::get<asset::SRasterizationParams>(protoPipeline)
+				);
+
+				return gpuRenderpassIndependentPipeline;
+			}
+
+			/*
+				Helper function for drawing full screen triangle.
+				It should be called between command buffer render pass
+				records.
+			*/
+
+			inline bool recordDrawCalls(video::IGPUCommandBuffer* commandBuffer, video::IGPUGraphicsPipeline* gpuGraphicsPipeline, const video::IGPUDescriptorSet *const* gpuDescriptorSets, uint32_t descriptorSetNumber = 3, uint32_t descriptorSetCount = 1)
+			{
+				_NBL_STATIC_INLINE_CONSTEXPR auto VERTEX_COUNT = 3;
+				_NBL_STATIC_INLINE_CONSTEXPR auto INSTANCE_COUNT = 1;
+
+				commandBuffer->bindGraphicsPipeline(gpuGraphicsPipeline);
+				commandBuffer->bindDescriptorSets(asset::EPBP_GRAPHICS, gpuGraphicsPipeline->getRenderpassIndependentPipeline()->getLayout(), descriptorSetNumber, descriptorSetCount, gpuDescriptorSets, nullptr);
+				return commandBuffer->draw(VERTEX_COUNT, INSTANCE_COUNT, 0, 0);
```

Comment:
dont bind desc set or pipeline

but make sure to unbind vertex/index bindings

Reply:
corrected